nixos/modules/services/matrix/meowlnir.nix

```diff
@@ -0,0 +1,309 @@
+              description = "The database URI.";
+            };
+          };
+          logging = {
+            min_level = lib.mkOption {
+              type = lib.types.str;
+              default = "debug";
+              description = "The minimum logging level.";
+            };
+            writers = lib.mkOption {
+              type = lib.types.listOf lib.types.attrs;
+              default = [
+                {
+                  type = "stdout";
+                  format = "json";
+                }
+              ];
+            };
+          };
+        };
+      };
+      example = {
+        homeserver = {
+          address = "http://localhost:8008";
+          domain = "example.com";
+        };
+
+        meowlnir = {
+          id = "meowlnir";
+          as_token = "$MEOWLNIR_AS_TOKEN";
+          hs_token = "$MEOWLNIR_HS_TOKEN";
+          pickle_key = "$MEOWLNIR_PICKLE_KEY";
+          management_secret = "$MEOWLNIR_MANAGEMENT_SECRET";
+          report_room = "!reportroom:example.com";
+        };
+
+        antispam.secret = "$MEOWLNIR_ANTISPAM_SECRET";
+
+        synapse_db = {
+          type = "postgres";
+          uri = "postgres://meowlnir:meowlnir@localhost/matrix-synapse?sslmode=disable";
+        };
+      };
+      default = { };
+      description = ''
+        {file}`config.yaml` configuration as a Nix attribute set.
+        Configuration options should match those described in
+        [example-config.yaml](https://github.com/maunium/meowlnir/blob/main/config/example-config.yaml).
+
+        Secret tokens should be specified using {option}`environmentFile`
+        instead
+      '';
+    };
+
+    registration = lib.mkOption rec {
+      apply = lib.recursiveUpdate default;
+      inherit (format) type;
+      example = {
+        as_token = "$MEOWLNIR_AS_TOKEN";
+        hs_token = "$MEOWLNIR_HS_TOKEN";
+        sender_localpart = "mohMex1ro0zaeraimeem";
+        namespaces = {
+          users = [
+            {
+              regex = "@abuse:example.com";
+              exclusive = true;
+            }
+          ];
+        };
+      };
+      default = {
+        inherit (cfg.settings.meowlnir) id;
+        url = cfg.settings.meowlnir.address;
+        sender_localpart = "meowlnirfakesenderlocalpart";
+        rate_limited = false;
+        "org.matrix.msc3202" = true;
+        "io.element.msc4190" = true;
+        "de.sorunome.msc2409.push_ephemeral" = true;
+        receive_ephemeral = true;
+      };
+      description = ''
+        {file}`registration.yaml` configuration as a Nix attribute set. See
+        [Appservice registration in the README](https://github.com/maunium/meowlnir/?tab=readme-ov-file#appservice-registration)
+
+        Secret tokens should be specified using {option}`environmentFile`
+        instead
+      '';
+    };
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
```

Comment:
`lib.types.path` is a bit of a footgun: when somebody places a paht literal here, the directory will be copied into the store (given we do `"${cfg.dataDir}"` below.

These days we have `pathWith { inStore = false; absolute = true; }` for this use-case.

This also applies to environmentFile.